SelfAndPeerEvaluations/utils.py

```diff
@@ -0,0 +1,439 @@
+import canvasapi
+from collections import defaultdict
+import csv
+from collections import deque
+import json
+import statistics
+import random
+
+
+class PartnerEvalQuizIndividualStats:
+    def __init__(self, name: str, id: int, files: list):
+        self.name = "".join(filter(str.isalnum, name))
+        self.id = str(id)
+        self.scores = defaultdict(list)
+
+        self.detailed_scores = []
+        self.qualitative_subjects = None
+
+        self.csv_file_path = (
+            f"./quiz_results/individual_student_reports/{self.name}.csv"
+        )
+        f = open(self.csv_file_path, "w")
+        self.writer = csv.writer(f)
+
+        self.__read_files(files)
+        self.__get_averages()
+        # transpose detailed scores
+        self.detailed_scores = [
+            [row[i] for row in self.detailed_scores]
+            for i in range(len(self.detailed_scores[0]))
+        ]
+        self.__format_csv_info()
+        self.writer.writerows(self.csv_file_info)
+        f.close()
```

Comment:
- This code looks like it could be moved to its own function.
- Prefer with statements when opening files instead of the manual open/close commands when possible